lib/components/home/artwork_rail.js

```diff
@@ -54,12 +56,18 @@ class ArtworkRail extends React.Component {
     // pop new view controller
   }
 
+  onGridLayout(event) {
+    this.setState({ gridHeight: event.nativeEvent.layout.height });
+  }
+
   renderModuleResults() {
     if (this.props.rail.results && this.props.rail.results.length) {
       return (
-        <View style={[styles.container, { height: this.state.expanded ? 2350 : 500 }]}>
-          <View style={[styles.gridContainer, { height: this.state.expanded ? 2300 : 420 }]}>
-            <Grid artworks={this.props.rail.results}/>
+        <View style={[styles.container, { height: this.state.expanded ? this.state.gridHeight + 150 : 500 }]}>
+          <View style={[styles.gridContainer, { height: this.state.expanded ? this.state.gridHeight + 100 : 420 }]}>
```

Comment:
The `gridContainer` contains the grid and also the View All button at the bottom when the rail is expanded.